### PR TITLE
update OpenAI API per docs 

### PR DESCRIPTION
**Pull request by nbsp**
includes changes we found that they didn't mention, such as add_message, generation_finished.

reverts the hotfix commit because i wanted a clean slate to update off of, but it pretty much reimplements all of it anyway.

index.js coming soon this is just the proto for now

### CODE REVIEW

File: plugins/openai/src/voice_assistant/index.ts [hunk 1]
```diff
@@ -25,22 +25,34 @@ import { AgentPlayout, type PlayoutHandle } from './agent_playout.js';
...
 
-export const defaultInferenceConfig: proto.InferenceConfig = {
+export const defaultSessionConfig: proto.SessionConfig = {
+  turn_detection: 'server_vad',
+  input_audio_format: proto.AudioFormat.PCM16,
+  transcribe_input: true,
+  vad: {
+    threshold: 0.5,
+    prefix_padding_ms: 500,
+    silence_duration_ms: 500,
```
Comment (bcherry) [inline plugins/openai/src/voice_assistant/index.ts]: i think the defaults are 300 and 200 - is this an intentional deviation?

File: plugins/openai/src/voice_assistant/index.ts [hunk 2]
```diff
@@ -265,6 +277,12 @@ export class VoiceAssistant {
       case proto.ServerEventType.INPUT_TRANSCRIBED:
         this.handleInputTranscribed(event);
         break;
+      case proto.ServerEventType.GENERATION_CANCELED:
+        this.handleModelListening();
```
Comment (bcherry) [inline plugins/openai/src/voice_assistant/index.ts]: rename this method

File: plugins/openai/src/voice_assistant/index.ts [hunk 3]
```diff
@@ -281,67 +299,67 @@ export class VoiceAssistant {
...
+  // XXX(nbsp): we use this event to determine which state to send, which breaks
+  // when more than one tool call can be processed at once. this needs to be rethought entirely.
+  //
+  // private handleAddMessage(event: proto.ServerEvent): void {
+  //   if (event.event !== proto.ServerEventType.ADD_MESSAGE) return;
+  //   if (event.message.content.type === 'tool_call') {
+  //     this.setState(proto.State.THINKING);
+  //     this.thinking = true;
+  //   }
+  // }
```
Comment (bcherry) [inline plugins/openai/src/voice_assistant/index.ts]: i'd just delete it and file a linear ticket to follow up

File: plugins/openai/src/voice_assistant/proto.ts [hunk 4]
```diff
@@ -219,18 +247,30 @@ export const NUM_CHANNELS = 1;
...
 export const OUTPUT_PCM_FRAME_SIZE = 1200; // 50ms
 
-export type InferenceConfig = {
+export type SessionConfig = Partial<{
+  turn_detection: 'disabled' | 'server_vad';
+  input_audio_format: AudioFormat;
+  transcribe_input: boolean;
+  vad: Partial<{
+    threshold: number; // 0..1 inclusive, default 0.5
+    prefix_padding_ms: number; // default 0.5
```
Comment (bcherry) [inline plugins/openai/src/voice_assistant/proto.ts]: default 300

File: plugins/openai/src/voice_assistant/index.ts (hunk 1 above)
Comment (nbsp) [reply plugins/openai/src/voice_assistant/index.ts]: oops no, fixed

**Pull request merged**